test/e2e/password_test.go

```diff
@@ -138,3 +138,217 @@ spec:
+
+		_, err := kubectl.RunWithOpts([]string{"delete", "secret", "password"},
+			RunOpts{AllowError: true})
+
+		if !strings.Contains(err.Error(), "(NotFound)") {
+			t.Fatalf("Expected NotFound error but was: %s", err)
+		}
+	})
+}
+
+func TestComplexPasswordTemplate(t *testing.T) {
+	env := BuildEnv(t)
+	logger := Logger{}
+	kapp := Kapp{t, env.Namespace, logger}
+	kubectl := Kubectl{t, env.Namespace, logger}
+
+	yaml1 := `
+---
+apiVersion: secretgen.k14s.io/v1alpha1
+kind: Password
+metadata:
+  name: password
+spec:  
+  length: 27
+  digits: 2
+  uppercaseLetters: 4
+  lowercaseLetters: 10
+  symbols: 3
+  secretTemplate:
+    type: Opaque
+    stringData:     
+      val: $(value)
+      
+`
+
+	name := "test-complex-password-template"
+	cleanUp := func() {
+		kapp.RunWithOpts([]string{"delete", "-a", name}, RunOpts{AllowError: true})
+	}
+
+	cleanUp()
+	defer cleanUp()
+
+	logger.Section("Deploy", func() {
+		kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", name},
+			RunOpts{IntoNs: true, StdinReader: strings.NewReader(yaml1)})
+	})
+
+	logger.Section("Check secret", func() {
+		out := waitForSecret(t, kubectl, "password")
+
+		var secret corev1.Secret
+
+		err := yaml.Unmarshal([]byte(out), &secret)
+		if err != nil {
+			t.Fatalf("Failed to unmarshal: %s", err)
+		}
+
+		if secret.Type != "Opaque" {
+			t.Fatalf("Wrong type")
+		}
+
+		if len(secret.Data["val"]) != 27 {
+			t.Fatalf("Expect password lenght is 27, found %d (value %s) ", len(secret.Data["val"]), secret.Data["val"])
+		}
+	})
+
+	logger.Section("Delete", func() {
+		kapp.Run([]string{"delete", "-a", name})
+
+		_, err := kubectl.RunWithOpts([]string{"delete", "secret", "password"},
+			RunOpts{AllowError: true})
+
+		if !strings.Contains(err.Error(), "(NotFound)") {
+			t.Fatalf("Expected NotFound error but was: %s", err)
+		}
+	})
+}
+
+func TestSymbolPasswordTemplate(t *testing.T) {
+	env := BuildEnv(t)
+	logger := Logger{}
+	kapp := Kapp{t, env.Namespace, logger}
+	kubectl := Kubectl{t, env.Namespace, logger}
+
+	yaml1 := `
+---
+apiVersion: secretgen.k14s.io/v1alpha1
+kind: Password
+metadata:
+  name: password
+spec:  
+  length: 3
+  symbols: 3
+  symbolCharSet: "!$#%"
+  secretTemplate:
+    type: Opaque
+    stringData:     
+      val: $(value)
+      
+`
+
+	name := "test-symbol-password-template"
+	cleanUp := func() {
+		kapp.RunWithOpts([]string{"delete", "-a", name}, RunOpts{AllowError: true})
+	}
+
+	cleanUp()
+	defer cleanUp()
+
+	logger.Section("Deploy", func() {
+		kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", name},
+			RunOpts{IntoNs: true, StdinReader: strings.NewReader(yaml1)})
+	})
+
+	logger.Section("Check secret", func() {
+		out := waitForSecret(t, kubectl, "password")
+
+		var secret corev1.Secret
+
+		err := yaml.Unmarshal([]byte(out), &secret)
+		if err != nil {
+			t.Fatalf("Failed to unmarshal: %s", err)
+		}
+
+		if secret.Type != "Opaque" {
+			t.Fatalf("Wrong type")
+		}
+
+		if len(secret.Data["val"]) != 3 {
```

Comment:
We use the `"github.com/stretchr/testify/assert"` package to test generally in SGC.

This would actually simply alot of this testing to something like:

```suggestion
		assert.Len(t, secret.Data["val"], 3, "Expect password length is 3, found %d", len(secret.Data["val"])")
```